src/main/java/rx/internal/operators/OperatorDoOnEmpty.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package rx.internal.operators;
+
+import rx.Observable;
+import rx.Subscriber;
+import rx.exceptions.Exceptions;
+import rx.functions.Action0;
+
+
+/**
+ * Perofrms an action of your choosing when the {@code Observable} has emitted no items after onCompleted() has been called
+ * this transformation as a new {@code Observable}.
+ * <p>
+ */
+public final class OperatorDoOnEmpty<T> implements Observable.Operator<T, T> {
+
+    private final Action0 action;
+
+    public OperatorDoOnEmpty(Action0 action) {
+        this.action = action;
+    }
+    @Override
+    public Subscriber<? super T> call(final Subscriber<? super T> child) {
+
+        return new Subscriber<T>() {
+
+            private volatile boolean isEmpty = true;
```

Comment:
Okay I knew this might come up and this topic always confuses me. But different threads can pass through the operator right? How is it guaranteed each thread will see the latest value change?

Not questioning the practice since all the other operators are designed without volatile. Bur just trying to figure out why its not needed...